### DISCUSSION TITLE
could one correlationId could only one record?

### DISCUSSION THREAD

**Discussion by tuohaibei** [elsa-workflows/elsa-core#4562] [Q&A]
I have many same correlationId records,beacause i execute many times by http endpoint.
How can i have only one record even though  execute many times by http endpoint.

![image](https://github.com/elsa-workflows/elsa-core/assets/5082315/efe7acda-d989-48a4-b29d-733fa77a91ed)

![image](https://github.com/elsa-workflows/elsa-core/assets/5082315/323400d2-b5be-4483-8e89-e68114b91785)

**Comment by tuohaibei** [comment]
@sfmskywalker hi,could you help me?

**Comment by sfmskywalker** [comment]
Hi @tuohaibei ,

If I understand correctly, you want to have only one workflow instance record, even when executing the workflow multiple times?

If yes, could you elaborate a little bit on why this is? Once I have a better understanding of the why, I might be in a better position to help you figure out a solution.

**Reply by tuohaibei** [reply]
Hi @sfmskywalker 
correlation ID I use the ID of our specific business, which is unique. Because of business needs, I want to operate on the same instance as long as it has the same correlation ID. Because we can call the business method associated with the same correlation ID multiple times in our business, but if this is placed in the workflow, multiple instances will be displayed, which is not very friendly.

**Reply by sfmskywalker** [reply]
I understand the challenge you're facing, and I appreciate your patience. In the current version of Elsa (Elsa 2), the functionality to operate on the same workflow instance based on a unique correlation ID isn't directly supported out of the box. However, I can suggest a workaround that might help you achieve the desired behavior:

1. **Initialize with a Dummy Activity**: Start your workflow with a "dummy" activity, such as a `WriteLine` activity. Connect this initial activity to your HTTP Endpoint activity. This setup ensures that your workflow is always ready to receive an HTTP request.

2. **Create a Loop**: Connect the last activity in your workflow back to the HTTP Endpoint activity. This creates a loop, allowing the workflow to handle multiple HTTP requests sequentially.

Please note that with this approach, you'll need to manually trigger the workflow once. After this initial trigger, the workflow will enter a continuous loop, awaiting incoming HTTP requests. 

To make this process a bit smoother, you might consider replacing the initial "dummy" activity with a duplicate of the HTTP Endpoint activity. Alternatively, you could try bypassing the additional activity altogether, though this might lead to ambiguity in determining the starting activity for the workflow engine.

Looking ahead, Elsa 3 is expected to provide more flexibility and control over specifying the starting activity of a workflow. This enhancement should simplify the process and offer a more straightforward solution to your use case.

Additionally, I've taken note of your requirement and will explore the possibility of introducing a setting in future Elsa versions to "reuse existing instance" based on the correlation ID. This feature could potentially provide a more elegant solution to your scenario.

I hope this workaround is helpful for now, and rest assured, we are continuously working to improve Elsa and appreciate your feedback and patience.

**Reply by tuohaibei** [reply]
Thank you @sfmskywalker ,i will try.thanks

**Reply by tuohaibei** [reply]
Hi @sfmskywalker 

I have already tried as you said,i have added  a duplicate of the HTTP Endpoint activity(chart1),
but when i request the http,correlate not executed(chart2).Do you have any suggestions?

i 
![image](https://github.com/elsa-workflows/elsa-core/assets/5082315/1b86468e-b545-4e37-b736-2656c6d70a56)

![image](https://github.com/elsa-workflows/elsa-core/assets/5082315/10ad07f9-54be-46f5-8250-5b7c9dc6fc7a)

**Reply by tuohaibei** [reply]
Can anyone help?

**Reply by sfmskywalker** [reply]
Hi @tuohaibei , make sure to make the HTTP request **twice** - the first time, you call it only to kick-start the workflow. When you make the HTTP request the second time, it should continue with the Correlate activity. If that doesn't happen, please let me know.

PS. Thank you for your sponsorship! 🙏🏻

**Reply by sfmskywalker** [reply]
OK, I see now the issue. You can't start your workflow with an HTTP Endpoint on the same route as the next HTTP Endpoint, because this will create both a trigger **and** a bookmark **using the same route and verb** - causing an ambiguity.

So instead of using the same paths, use e.g. `/start` for the first HTTP endpoint or use a different activity like WriteLine and start the workflow manually using the REST API. That will create one workflow instance, to which you can send HTTP requests infinitely.

Here's a small sample: [workflow-definition-38ff9c5632c1487f93b150505746a123.json.zip](https://github.com/elsa-workflows/elsa-core/files/13242219/workflow-definition-38ff9c5632c1487f93b150505746a123.json.zip)

When you import that workflow, you need to make an initial HTTP request to /workflows/start and then you can continue making HTTP request to /workflows/my-api - you'll notice that no new instances will be created.
